storage/innobase/fil/fil0fil.cc

```diff
@@ -4847,17 +4815,15 @@ fil_node_complete_io(fil_node_t* node, const IORequest& type)
 		ut_ad(!srv_read_only_mode
 		      || fsp_is_system_temporary(node->space->id));
 
-		++fil_system->modification_counter;
-
-		node->modification_counter = fil_system->modification_counter;
+		node->needs_flush = true;
 
 		if (fil_buffering_disabled(node->space)) {
 
 			/* We don't need to keep track of unflushed
 			changes as user has explicitly disabled
 			buffering. */
 			ut_ad(!node->space->is_in_unflushed_spaces);
-			node->flush_counter = node->modification_counter;
+			node->needs_flush = false;
```

Comment:
I think that we should remove both assignments to `node->needs_flush` above, and instead assert `ut_ad(!node->needs_flush)` inside the `if`.

Reply:
At least one `node->needs_flush` is needed somewhere :)
Changed this code in the last commit.